src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1333,9 +1333,51 @@ def post_process(self, r):
                 while r is not None and r["properties"]["provisioningState"].lower() == "inprogress":
                     r = self.client.show(self.cmd, self.get_argument_resource_group_name(), self.get_argument_name())
                     time.sleep(1)
-                linker_client.linker.begin_create_or_update(resource_uri=r["id"],
-                                                            parameters=item["parameters"],
-                                                            linker_name=item["linker_name"]).result()
+
+                service_connectors_def_list, service_bindings_def_list = parse_service_bindings(
+                    self.cmd, self.get_argument_service_bindings(), self.get_argument_resource_group_name(), self.get_argument_name())
+                self.set_argument_service_connectors_def_list(
+                    service_connectors_def_list)
+
+                # Case: Kafka on confluent cloud with two linker operations.
+                # Validate uniqueness across all bindings
+                if any(len(item["parameters"]) ==
+                       2 for item in service_connectors_def_list):
+                    updated_service_connectors = update_connectors_with_two_parameters(
+                        item, service_connectors_def_list.copy(), r["id"])
+                    check_bindings_and_raise_error(
+                        self.cmd,
+                        updated_service_connectors,
+                        service_bindings_def_list,
+                        self.get_argument_resource_group_name(),
+                        self.get_argument_name())
+                else:
+                    check_bindings_and_raise_error(
+                        self.cmd,
+                        service_connectors_def_list,
+                        service_bindings_def_list,
+                        self.get_argument_resource_group_name(),
+                        self.get_argument_name())
+
+                if len(item["parameters"]) == 1:
+                    linker_create_or_update(
+                        linker_client=linker_client,
+                        r=r,
+                        parameters=item["parameters"][0],
+                        linker_name=item["linker_name"])
+                else:
+                    parameters_bootstrap_server, parameters_schema_registry = item["linker_name"].split(
+                        '.', 1)
+                    linker_create_or_update(
+                        linker_client=linker_client,
+                        r=r,
+                        parameters=item["parameters"][0],
+                        linker_name=parameters_bootstrap_server)
+                    linker_create_or_update(
+                        linker_client=linker_client,
+                        r=r,
+                        parameters=item["parameters"][1],
+                        linker_name=parameters_schema_registry)
```

Comment:
why do we need to call the function one by one?

Reply:
it's not needed really. i will split the linker_name into a list and that should give iterable indices for both parameters and linker_name. good catch!